apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/PromptConfiguration/BehaviourSettings.tsx

```diff
@@ -22,6 +23,22 @@ export function BehaviourSettings({
     defaultValue: undefined,
   })
 
+  const {
+    value: disabledAgentOptimization,
+    setValue: setDisableAgentOptimization,
+  } = useConfigValue<boolean>({
+    config,
+    setConfig,
+    key: 'disableAgentOptimization',
+    defaultValue: false,
+  })
+
+  useEffect(() => {
+    if (agentValue !== 'agent') {
+      setDisableAgentOptimization(undefined)
```

Comment:
Instead of this do this logic on the type callback function. Avoid `useEffect` like la peste 🙏